File: src/core/models/container.js
"use strict";

module.exports = function(sequelize, DataTypes) {

    var Container = sequelize.define("Container", {
        id: {
            type: DataTypes.STRING(128),
            primaryKey: true
        },
        name: DataTypes.STRING(128),
        volumes: DataTypes.STRING(256),
        ports: DataTypes.STRING(256),
        public: DataTypes.STRING(8),
        image: DataTypes.STRING(128),
        domain: DataTypes.STRING(128),
        state: DataTypes.STRING(128),
        metadata: DataTypes.TEXT,
        configuration: DataTypes.TEXT
    },{
        tableName: 'containers',

        updatedAt: 'last_update',
        createdAt: 'date_of_creation'
    });

    return Container;
};

File: db/tables/container.go
package tables

import (
	"time"
)

type Container struct {
	ID        string `gorm:"primary_key" sql:"type:varchar(64)"`
	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt *time.Time

	Image  string `sql:"type:varchar(255);not null"`
	Status string `sql:"type:varchar(255);not null"`
}
